lib/providers/gcm.js

```diff
@@ -79,13 +79,16 @@ GcmProvider.prototype._createMessage = function(notification) {
     delayWhileIdle: notification.delayWhileIdle,
   });
 
-  var propNames = Object.keys(notification);
-  // GCM does not have reserved message parameters for alert or badge, adding them as data.
-  propNames.push('alert', 'badge');
+  // Dont send these keys in the message payload
+  var exclusions = ['id', 'deviceToken', 'deviceType']
 
-  propNames.forEach(function(key) {
+  // We need to use data in order to get additional payload
+  Object.keys(notification.__data)
```

Comment:
AFAIK, `__data` is containing only the properties that are specified in model definition or loaded from the database. It's possible to set a model property that won't be stored in `__data`, for example by calling the following code: `notification.extraData = 'value';`

Instead of iterating over `__data`, we should call `.toObject()` (or `.toJSON()`) to get a plan javascript object representing model data.